### ISSUE TITLE
[Bug]: [SearchBox] "Clear Text" button cannot be reach through keyboard navigation

### ISSUE THREAD

**Issue by snisonoffMS** [microsoft/fluentui#33895]
### Component

SearchBox

### Package version

9.58.3

### React version

^16.14.0

### Environment

```shell
System:
    OS: Windows 11 10.0.26120
    CPU: (12) x64 Intel(R) Xeon(R) W-2235 CPU @ 3.80GHz
    Memory: 37.09 GB / 63.73 GB
  Browsers:
    Edge: Chromium (133.0.3065.51)
```

### Current Behavior

It is not possible to navigate to the clear text button with keyboard navigation

### Expected Behavior

Since the clear text button is actionable, it should be navigable through keyboard navigation

### Reproduction

https://stackblitz.com/edit/x26suabm?file=src%2Fexample.tsx

### Steps to reproduce

1. Run https://stackblitz.com/edit/x26suabm?file=src%2Fexample.tsx
2. navigate into the searchbox
3. tab
4. notice that the clear text button is skipped

### Are you reporting an Accessibility issue?

yes

### Suggested severity

Urgent - No workaround and Products/sites are affected

### Products/sites affected

TwoUI

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by miroslavstastny** [comment]
This is by design: [packages/react-components/react-search/library/docs/Spec.md](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-search/library/docs/Spec.md#accessibility)
> The clear button is not in the tab order and should not recieve focus
> Keyboard users can use keyboard shortcuts to clear the value.
> The clear button is exposed to voice control and screen reader users.

fyi @microsoft/cxe-prg
